code/game/objects/items/weapons/RMS_vr.dm

```diff
@@ -0,0 +1,231 @@
+
+/obj/item/weapon/rms/Initialize()
+	. = ..()
+	src.spark_system = new /datum/effect/effect/system/spark_spread
+	spark_system.set_up(5, 0, src)
+	spark_system.attach(src)
+	update_icon()
+
+/obj/item/weapon/pipe_dispenser/Destroy()
+	qdel_null(spark_system)
+	return ..()
+
+/obj/item/weapon/rms/update_icon()
+	charge_stage = round((stored_charge/max_charge)*4)
+	if(charge_stage >= 4)
+		charge_stage = 4
+	add_overlay("rms_charge[charge_stage]")
+
+/obj/item/weapon/rms/examine(mob/user)
+	. = ..()
+	. += display_resources()
+
+/obj/item/weapon/rms/proc/display_resources()
+	return "It currently holds [round(stored_charge/1000)]/[max_charge/1000] kW charge."
+
+/obj/item/weapon/rms/proc/drain_battery(user, battery)
+	var/obj/item/weapon/cell.C = battery
+	if(stored_charge == max_charge)
+		to_chat(user, "<span class='notice'>The Rapid Material Synthesizer is full on charge!.</span>")
+		update_icon()
+		return
+	if(C.charge == 0)
+		to_chat(user, "<span class='notice'>The battery has no charge.</span>")
+		update_icon()
+		return
+	else
+		playsound(get_turf(src), 'sound/machines/click.ogg', 50, 1)
+		if(do_after(user, 2,target = C))
+			stored_charge += C.charge
+			C.charge = 0
+			C.update_icon()
+			to_chat(user, "<span class='notice'>You drain [C].</span>")
+			if(stored_charge > max_charge)
+				stored_charge = max_charge
+			update_icon()
+			return
+
+/obj/item/weapon/rms/proc/consume_resources(amount)
+	stored_charge -= amount
+	update_icon()
+	return
+
+
+/obj/item/weapon/rms/proc/can_afford(amount)
+	if(mode_index != modes.len)
+		if(stored_charge >= charge_cost)
+			return TRUE
+		else
+			return FALSE
+	else
+		if(stored_charge >= charge_cost_r)
+			return TRUE
+		else
+			return FALSE
+
+/obj/item/weapon/rms/proc/use_rms(atom/A, mob/living/user)
+	var/obj/product
+	if(!can_afford())
+		to_chat(user, "<span class='notice'>There is not enough charge to use this mode.</span>")
+		return
+	playsound(src.loc, 'sound/machines/click.ogg', 50, 1)
+	if(do_after(user, 5, target = A))
+		switch(mode_index)
+			if(1)
+				if(charge_stage >= 4)
+					product = new /obj/item/stack/material/plasteel  // can only create one sheet at max charge, and uses five times more charge
+					consume_resources(charge_cost_m)
+					spark_system.start()
+				else
+					product = new /obj/item/stack/material/steel
+					consume_resources(charge_cost)
+			if(2)
+				if(charge_stage >= 4)
+					product= new /obj/item/stack/material/glass/phoronglass // can only create one sheet at max charge, and uses five times more charge
+					consume_resources(charge_cost_m)
+					spark_system.start()
+				else
+					product = new /obj/item/stack/material/glass
+					consume_resources(charge_cost)
+			if(3)
+				switch(charge_stage)
+					if(4)
+						product = pick(50; new /obj/item/weapon/cell/secborg/empty,
+										20; new /obj/item/weapon/cell/high/empty,
+										15; new /obj/item/weapon/cell/super/empty,
+										10; new /obj/item/weapon/cell/hyper/empty,
```

Comment:
I dunnooooo about hyper cells, or even cells in general. I mean power is essentially infinite on the station most of the time, so giving someone the ability to repeatedly put a cell into a charger, then charge the RMS, then repeatedly create other cells without science being involved.

Reply:
I gotta say I agree.

Reply:
I would like to bring into question why Science is perceived to be in charge of "power" when Engineering is the one that provides it.  I mostly made this feature since I personally believe that Engineering should have some influence in providing storage for power, with a probability of an upgrade, its just gated by total charge of the device.

I was tempted to make a random machine part feature since Engineering does get the parts lathe circuit board, but I think that would actually affect Science directly, since machine parts directly correlate with upgrades, and I definitely do not want to do that.

Something that I would like to personally steer Engineering towards is not only their production of power, but having some incentive to benefit from the power they create, this is the main purpose of the device.  A secondary effect would also be monitoring the drain of batteries on the grid to give Engineering a reason to revisit the RCON more than once aside from checking for grubs and other anomalies that usually happen in Science.

I can easily make the cost of creating more batteries higher, since "repeatedly creating other cells" seems to be a primary concern, and I can see that myself, but given that hypers can be found in junk piles, I would like to keep this probability in tact at least.